### DISCUSSION TITLE
Build in login for VNC window

### DISCUSSION THREAD

**Discussion by djniviro** [apohl79/audiogridder#1097] [Ideas]
Hi there,

One other, quite useful feature request from my side:

Currently I have to open up a VNC connection via a remote desktop app to run the AudioGridder visual plugins. Otherwise the Agridder VST freezes. Is it possible adding, on the local machine, a VNC login page so I don't have to waste a whole monitor for sharing the screen for AudioGridder only?

Also, a nice feature would be a windowed monitoring tab of the connections on the local machine. Now I have to show it and it disappears after a couple of seconds again. It's also not possible to drag it around.

Thank you in advance for being awesome and creating this!

**Comment by apohl79** [comment]
Maybe you can use an HDMI dummy. This would remove the need for VNC.

> a windowed monitoring tab of the connections on the local machine

Not sure I get how this is supposed to work. The AG plugin menu shows all the connections. Could you explain how you would like the feature to behave?

> It's also not possible to drag it around.

I guess you are talking about the plugin monitor window that should be movable?

**Reply by djniviro** [reply]
That dummy might indeed be a viable option! I also noticed that, when I have opened up my Remote Desktop connection on a dedicated screen, the movements of the plugins and mouse is far more smooth than with the AudioGridder plugin view, even though it uses the same content. Any tips to improve latency and framerate on the AudioGridder local VST plugin?

Well for now the AG connections menu/plugin monitor is either an on request or a pop-up window when something's getting disconnected, fix mounted on the right upper corner of the main screen. I'd love to see a feature in the future where you could open it up windowed like a dedicated application which you could move around (to other screens) and have it open when working on the project.

Once again thank you so much for the amazing tool you're creating. Can't live without anymore!

Best!

**Reply by apohl79** [reply]
> I also noticed that, when I have opened up my Remote Desktop connection on a dedicated screen, the movements of the plugins and mouse is far more smooth than with the AudioGridder plugin view, even though it uses the same content. 

I think VNC has a better/more efficient way of capturing the screen content (and transmitting it). AG uses either avfoundation on macos or gdi on windows. At some point it might be worth using VNC for this.

> Any tips to improve latency and framerate on the AudioGridder local VST plugin?

If the server is on macos, do not use a retina resolution and use the smallest resolution that works for you. This is because avfoundation always captures the full screen and AG is cropping it down to the plugin window. With retina the images are pretty big.

On windows scaling should be disabled.

In the AG server settings, you can try a lower quality.

> Well for now the AG connections menu/plugin monitor is either an on request or a pop-up window when something's getting disconnected, fix mounted on the right upper corner of the main screen. I'd love to see a feature in the future where you could open it up windowed like a dedicated application which you could move around (to other screens) and have it open when working on the project.

Ok. Already added a FR for the movable monitor window. Will add the connection view as another FR. Thx for the suggestion!

> Once again thank you so much for the amazing tool you're creating. Can't live without anymore!

Cheers :-)
